### PR TITLE
Make Π do the same as Þ* when not given a list of numbers

### LINKED ISSUE #757: Merge `Þ*` into `Π`
`Π` is the product builtin, reducing an integer list by multiplication and doing not much else for string lists / list lists. `Þ*` reduces a list by cartesian product.

These seem perfect to merge - both have dubious, mostly useless behaviour on unintended inputs, and they're quite similar - reduce by cartesian product vs take product.

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #757

### CODE REVIEW

File: vyxal/elements.py [hunk 1]
```diff
@@ -2964,8 +2964,14 @@ def prepend(lhs, rhs, ctx):
 
 def product(lhs, ctx):
     """Element Π
-    (lst) -> product(list)"""
-    return vy_reduce(multiply, lhs, ctx=ctx)
+    (lst[num]) -> product(list)
+    (lst[str|lst]) -> Cartesian product over a list of lists
+    """
+
+    if all(vy_type(x) == NUMBER_TYPE for x in lhs):
```
Comment (cgccuser) [inline vyxal/elements.py:2971]: This will hang for infinite lists

**Review by cgccuser — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 1 above)
Comment (chunkybanana) [reply vyxal/elements.py:2971]: Neither product nor reduce by cartesian product make any sense for an infinite list. Doesn't matter.

**Review by chunkybanana — COMMENTED** [review]
String testcase please?

**Review by chunkybanana — COMMENTED** [review]
Actually, can we make it behave like regular cartesian product on strings, and turn a string list into a string list?

File: vyxal/elements.py (hunk 1 above)
Comment (cgccuser) [reply vyxal/elements.py:2971]: You right

File: documents/knowledge/elements.yaml [hunk 2]
```diff
@@ -4676,17 +4681,6 @@
     - "[[1, 2, 3, 4, 5, 6]] : [[1, 2, 3, 4, 5, 6], [2, 3, 4, 5, 6], [3, 4, 5, 6], [4, 5, 6], [5, 6], [6]]"
     - "[[]] : []"
 
-
```
Comment (cgccuser) [inline documents/knowledge/elements.yaml:4679]: Wait why was this removed ?

**Review by cgccuser — CHANGES_REQUESTED** [review]
Looks good but why was PI removed and not P*?

File: documents/knowledge/elements.yaml (hunk 2 above)
Comment (Lyxal) [reply documents/knowledge/elements.yaml:4679]: It wasn't. That's the result of github branch frickery

**Review by chunkybanana — APPROVED** [review]

**Review by cgccuser — APPROVED** [review]
